File: features/master/src/main/java/com/nivasafinance/features/master/pincode/exception/PincodeOperationException.java
package com.nivasafinance.features.master.pincode.exception;

import com.nivasafinance.common.exception.ExceptionUtils;
import org.springframework.context.MessageSource;

import java.io.Serial;

public class PincodeOperationException extends RuntimeException {
    
    @Serial
    private static final long serialVersionUID = 1L;
    
    public PincodeOperationException(String messageKey, MessageSource messageSource) {
        super(ExceptionUtils.createLocalizedMessage(messageKey, null, messageSource));
    }
}

File: features/master/src/main/java/com/nivasafinance/features/master/pincode/exception/PincodeOperationException.kt
package com.nivasafinance.features.master.pincode.exception

import com.nivasafinance.common.exception.ExceptionUtils
import org.springframework.context.MessageSource

class PincodeOperationException(
    messageKey: String,
    messageSource: MessageSource
) : RuntimeException(
    ExceptionUtils.createLocalizedMessage(messageKey, null, messageSource)
)
